libc/include/sys/queue.h

```diff
@@ -96,27 +148,61 @@
     struct type *next;                                                         \
   }
 
+#define STAILQ_CLASS_ENTRY(type)                                               \
+  struct {                                                                     \
+    class type *next;                                                          \
+  }
+
 // Singly-linked tail queue access methods.
 
 #define STAILQ_EMPTY(head) ((head)->first == NULL)
 #define STAILQ_FIRST(head) ((head)->first)
+#define STAILQ_LAST(head, type, field)                                         \
+  STAILQ_EMPTY(head)                                                           \
+  ? NULL                                                                       \
+  : (QUEUE_TYPEOF(type) *)((char *)(head)->last -                              \
+                           offsetof(QUEUE_TYPEOF(type), field))
```

Comment:
as you define `offsetof`, providing a `containerof` for us would make this slightly less gross.

Reply:
Done.